client/src/modules/user/AuthBox/components/PasswordChangeForm.jsx

```diff
@@ -0,0 +1,293 @@
+    );
+  };
+
+  onNewPasswordChange = (newPassword, isValid) => {
+    const {
+      higherLevelErrors,
+      higherLevelErrors: { newPasswordError }
+    } = this.state;
+    const error = isValid ? '' : newPasswordError;
+
+    this.setState(
+      {
+        higherLevelErrors: {
+          ...higherLevelErrors,
+          newPasswordError: error
+        },
+        newPassword,
+        isNewPasswordValid: isValid
+      },
+      this.comparePasswords
+    );
+  };
+
+  onPasswordConfirmChange = (confirmPasswordValue, isValid) =>
+    this.setState(
+      { confirmPasswordValue, isPasswordConfirmValid: isValid },
+      this.comparePasswords
+    );
+
+  passwordValidator = value => {
+    if (!validatePassword(value)) {
+      return 'user.auth.input.password.invalid';
+    }
+
+    return '';
+  };
+
+  comparePasswords = () => {
+    const {
+      higherLevelErrors,
+      higherLevelErrors: { confirmPasswordValueError },
+      newPassword,
+      confirmPasswordValue
+    } = this.state;
+    let newError;
+
+    if (confirmPasswordValueError && newPassword === confirmPasswordValue) {
+      newError = '';
+    }
+
+    if (
+      !confirmPasswordValueError &&
+      confirmPasswordValue !== undefined &&
+      newPassword !== confirmPasswordValue
+    ) {
+      newError = 'user.auth.input.password.not-match';
+    }
+
+    if (newError !== undefined) {
+      this.setState({
+        higherLevelErrors: {
+          ...higherLevelErrors,
+          confirmPasswordValueError: newError
+        }
+      });
+    }
+  };
+
+  isFormValid = () => {
+    const {
+      higherLevelErrors,
+      isNewPasswordValid,
+      isPasswordConfirmValid,
+      isPasswordValid
+    } = this.state;
+    const isError = _some(higherLevelErrors, error => error !== '');
+
+    return this.setState({
+      isFormValid:
+        isNewPasswordValid &&
+        isPasswordConfirmValid &&
+        isPasswordValid &&
+        !isError
+    });
+  };
+
+  handleChangePassword = event => {
+    event.preventDefault();
+    const { password, newPassword, confirmPasswordValue } = this.state;
+
+    this.setState({ pending: true });
+
+    this.pendingPromise = makeAbortablePromise(
+      changePassword(password, newPassword, confirmPasswordValue)
+    );
+
+    return this.pendingPromise.promise
+      .then(() => this.setState({ passwordChanged: true }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          const newState = { pending: false };
+
+          if (err instanceof ValidationException) {
+            const { isConfirmPasswordError, isNewPasswordError } = err.errors;
+            newState.higherLevelErrors = {
+              confirmPasswordValueError: isConfirmPasswordError
+                ? 'user.auth.input.password.not-match'
+                : '',
+              newPasswordError: isNewPasswordError
+                ? 'user.auth.input.password.invalid'
+                : ''
+            };
```

Comment:
add return here and else will be not needed

Reply:
`if-else` isn't an evil ;-). Here we check if ValidationException or any other exception so I think `if-else` is adequate. But there is a reason that I can't use a `return` here. After this `if-else` we set the state and the `return` in `if`  will break this `catch` and the code after `if-else` won't' be performed.